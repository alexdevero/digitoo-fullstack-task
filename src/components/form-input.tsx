import { Dispatch, SetStateAction } from 'react'

interface FormInputUI {
  inputId: string;
  inputLabel: string;
  handleInputChange: Dispatch<SetStateAction<string>>;
}

export const FormInput = (props: FormInputUI) => (
  <div className="form-group">
    <label htmlFor={props.inputId} className="small">{props.inputLabel}</label>

    <input
      type="text"
      className="form-control"
      id={props.inputId}
      onChange={(event) => props.handleInputChange(event.currentTarget.value)}
    />
  </div>
)
