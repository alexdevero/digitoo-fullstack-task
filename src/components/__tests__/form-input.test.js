import React from 'react'
import renderer from 'react-test-renderer'

import { FormInput } from './../form-input'

it('Form input renders correctly', () => {
  const tree = renderer
    .create(<FormInput inputLabel="String 1" inputType="str1" inputId="str1" handleInputChange={() => {}} />)
    .toJSON()

  expect(tree).toMatchSnapshot()
})
