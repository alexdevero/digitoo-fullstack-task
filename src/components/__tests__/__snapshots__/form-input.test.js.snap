// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form input renders correctly 1`] = `
<div
  className="form-group"
>
  <label
    className="small"
    htmlFor="str1"
  >
    String 1
  </label>
  <input
    className="form-control"
    id="str1"
    onChange={[Function]}
    type="text"
  />
</div>
`;
